apps/cowswap-frontend/src/theme/ThemedGlobalStyle.tsx

```diff
@@ -73,7 +73,7 @@ export const ThemedGlobalStyle = createGlobalStyle`
 
   // TODO: Can be removed once we control this component
   [data-reach-dialog-overlay] {
-    z-index: 10 !important;
+    z-index: 99999 !important;
```

Comment:
Hardcoded z-index with an`!important` is usually a red flag. There's also the `TODO` above

Is this really needed? 

If it passes our QAs I'm fine with this, but probably we can handle this better. Maybe here https://github.com/cowprotocol/cowswap/blob/4859e6f9c538ff39001b753195e07f1533f0872f/apps/cowswap-frontend/src/common/pure/Modal/styled.tsx#L76

Reply:
Got it 👍 Looks like this change causing some more issues in other places which is not super ideal.